packages/terra-status-view/src/StatusView.jsx

```diff
@@ -36,223 +43,146 @@ const propTypes = {
-      }
-    } else if (!this.props.isAlignedTop) {
-      // glyph does not exist or does not fit, distribute remaining padding if any to center
-      const remainingHeight = viewHeight - contentHeight;
-      if (remainingHeight > 0) {
-        paddingTop = remainingHeight * 0.4;
-        paddingBottom = remainingHeight * 0.6;
-      }
-    }
-
-    // set which components can fit into the view
-    this.setState({
-      showGlyph,
-      paddingTop,
-      paddingBottom,
-    });
-  }
-
-  render() {
-    const {
-      children,
-      customGlyph,
-      isAlignedTop,
-      isGlyphHidden,
-      message,
-      title,
-      variant,
-      ...customProps
-    } = this.props;
-
-    const { intl } = this.context;
-
-    let glyphSection;
-    if (!isGlyphHidden && this.state.showGlyph) {
-      if (variant === StatusViewVariants.CUSTOM) {
-        glyphSection = (
-          <div className={cx('glyph')} ref={(element) => { this.glyphNode = element; }}>
-            {customGlyph}
-          </div>
-        );
-      } else {
-        glyphSection = (
-          <div className={cx('glyph')} ref={(element) => { this.glyphNode = element; }}>
-            <svg className={cx(variant)} />
-          </div>
-        );
-      }
-    }
+  const generatedButtonsArray = [];
+  buttonAttrsArray.forEach(button => generatedButtonsArray.push(
+    <Button {...button} className={cx(['button', button.className])} />,
+  ));
+  return generatedButtonsArray;
+};
 
-    let messageSection;
-    if (message) {
-      messageSection = (
-        <div className={cx('message')} ref={(element) => { this.messageNode = element; }}>
-          {message}
+const StatusView = ({
+  buttonAttrs,
+  customGlyph,
+  intl,
+  isAlignedTop,
+  isGlyphHidden,
+  message,
+  title,
+  variant,
+  ...customProps
+}) => {
+  let glyphSection;
+  if (!isGlyphHidden) {
+    if (customGlyph) {
+      glyphSection = (
+        <div className={cx('glyph')}>
+          {customGlyph}
         </div>
       );
-    }
-
-    let actionSection;
-    if (React.Children.count(children) > 0) {
-      actionSection = (
-        <div className={cx('actions')} ref={(element) => { this.actionsNode = element; }}>
-          {children}
+    } else {
+      glyphSection = (
+        <div className={cx('glyph')}>
+          <svg className={cx(variant)} />
         </div>
       );
     }
+  }
 
-    let defaultTitle = title;
-    if (!defaultTitle) {
-      defaultTitle = (variant === StatusViewVariants.CUSTOM) ? '' : intl.formatMessage({ id: `Terra.status-view.${variant}` });
-    }
-
-    let dividerSection;
-    if (messageSection || actionSection) {
-      dividerSection = (
-        <div className={cx('divider')} ref={(element) => { this.dividerNode = element; }}>
-          <Divider />
-        </div>
-      );
-    }
+  let defaultTitle;
+  if (checkVariantExistence(variant)) {
+    defaultTitle = intl.formatMessage({ id: `Terra.status-view.${variant}` });
```

Comment:
All things considered, you could probably use react-intl's `<FormattedMessage />` component here and remove `injectIntl` and the prop. It'll wrap the string in a <span> tag, but I don't think that will impact anything.

Reply:
updated to FormattedMessage